Tests/IntegrationTests/Todo.WebApi.IntegrationTests/Infrastructure/TestWebApplicationFactory.cs

```diff
@@ -48,6 +55,49 @@ protected override void ConfigureWebHost(IWebHostBuilder builder)
             });
         }
 
+        public async Task<HttpClient> CreateClientWithJwtToken()
+        {
+            string accessToken = await GetAccessToken().ConfigureAwait(false);
+            HttpClient httpClient = CreateClient();
+            httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {accessToken}");
+            return httpClient;
+        }
+
+        private async Task<string> GetAccessToken()
+        {
+            IConfiguration configuration = base.Services.GetRequiredService<IConfiguration>();
+            string requestUri = $"https://{configuration["Auth0:Domain"]}/oauth/token";
+            string audience = configuration["Auth0:Audience"];
+            string clientId = configuration["Auth0:ClientId"];
+            string clientSecret = configuration["Auth0:ClientSecret"];
+
+            using (HttpClient httpClient = base.Services.GetRequiredService<IHttpClientFactory>().CreateClient())
+            {
+                var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, requestUri)
+                {
+                    Content = new FormUrlEncodedContent(new List<KeyValuePair<string, string>>
+                    {
+                        new KeyValuePair<string, string>("grant_type", "client_credentials"),
+                        new KeyValuePair<string, string>("client_id", clientId),
+                        new KeyValuePair<string, string>("client_secret", clientSecret),
+                        new KeyValuePair<string, string>("audience", audience)
+                    })
+                };
+                HttpResponseMessage httpResponseMessage =
+                    await httpClient.SendAsync(httpRequestMessage).ConfigureAwait(false);
+
+                if (!httpResponseMessage.IsSuccessStatusCode)
+                {
+                    throw new Exception(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/satrapu/aspnet-core-logging/pullRequest?prid=5148460)